src/secrets.js

```diff
@@ -72,7 +72,21 @@ async function getSecrets(secretRequests, client) {
  */
 async function selectData(data, selector) {
     const ata = jsonata(selector);
-    let result = JSON.stringify(await ata.evaluate(data));
+    let d = await ata.evaluate(data);
+
+    // If we have a Javascript Object, then this data was stored in Vault as
+    // pure JSON (not a JSON string)
+    const storedAsJSONData = isObject(d);
+
+    if (isJSONString(d)) {
+        // If we already have a JSON string we will not "stringify" it yet so
+        // that we don't end up calling JSON.parse. This would break the
+        // secrets that are stored as pure JSON. See: https://github.com/hashicorp/vault-action/issues/194
```

Comment:
I feel like you mentioned this before so apologies if this is a repeated q:
Are we avoiding calling `JSON.stringify` here because under the hood it calls `JSON.parse`? In that case, a follow up — Hasn't the try-catch block on [L139-L143](https://github.com/hashicorp/vault-action/pull/473/files#diff-47be61e4446c15650db3c077e509c50ccfa2b4626c65fe43bad550fe25e611a9R139-R143) ensured that `JSON.parse` does not break things?

Reply:
No, JSON.parse does not call JSON.stringify --if I am understanding you correctly. I see how this comment might convey that though. I will see if I can improve the wording a bit.